frontend/src/metabase/parameters/components/ParameterLinkedFilters/use-filter-fields.ts

```diff
@@ -0,0 +1,60 @@
+import { useCallback, useState } from "react";
+import { t } from "ttag";
+import { DashboardApi } from "metabase/services";
+import { useOnMount } from "metabase/hooks/use-on-mount";
+import { UiParameter } from "metabase-lib/parameters/types";
+
+export interface UseFilterFieldsState {
+  data?: string[][];
+  error?: string;
+  loading: boolean;
+}
+
+const useFilterFields = (
+  parameter: UiParameter,
+  otherParameter: UiParameter,
+): UseFilterFieldsState => {
+  const [state, setState] = useState<UseFilterFieldsState>({ loading: false });
+
+  const handleLoad = useCallback(async () => {
+    const filtered = getParameterFieldIds(parameter);
+    const filtering = getParameterFieldIds(otherParameter);
+
+    if (!filtered.length || !filtered.length) {
```

Comment:
Doesn't look intentional — these two expressions seem to be identical

Reply:
Great catch!